src/test/java/com/wikia/webdriver/testcases/auth/FacebookTests.java

```diff
@@ -0,0 +1,76 @@
+package com.wikia.webdriver.testcases.auth;
+
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.RelatedIssue;
+import com.wikia.webdriver.common.core.api.GraphApi;
+import com.wikia.webdriver.common.core.helpers.FacebookUser;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.modalwindows.FacebookSignupModalComponentObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.register.AttachedRegisterPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.register.DetachedRegisterPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.signin.AttachedSignInPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.facebook.FacebookMainPageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.special.preferences.PreferencesPageObject;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = {"auth-facebook"})
+public class FacebookTests extends NewTestTemplate {
+
+  private GraphApi facebookApi = new GraphApi();
+  private FacebookUser facebookUser;
+
+  @BeforeSuite
+  private void setUp(){
+    facebookUser = facebookApi.createFacebookTestUser();
+  }
+
+  @AfterSuite
+  private void cleanUp() {
+    facebookApi.deleteFacebookTestUser(facebookUser.getId());
+  }
+
+  public void facebookButtonIsVisibleOnSignUpPage() {
+    AttachedRegisterPage registerPage = new WikiBasePageObject().openSpecialUserSignUpPage(wikiURL);
+    assertTrue(registerPage.isConnectWithFacebookButtonVisible());
+  }
+
+  public void facebookButtonIsVisibleOnLoginPage() {
+    AttachedSignInPage signInPage = new WikiBasePageObject().openSpecialUserLogin(wikiURL);
+    assertTrue(signInPage.isConnectWithFacebookButtonVisible());
+  }
+
+  public void facebookButtonIsVisibleOnForcedLoginModal() {
```

Comment:
These look like helper methods, should they be `public`?

Reply:
these are test methods - `TestNG` picks up all public methods that return void and executes them as tests. They're not marked with `@Test` because the whole class has that annotation.